mods/3d_armor/shields/init.lua

```diff
@@ -30,18 +71,34 @@ if ARMOR_MATERIALS.wood then
 	})
 end
 
-if ARMOR_MATERIALS.cactus then
-	minetest.register_tool("shields:shield_cactus", {
-		description = "Cactus Shield",
+if armor.materials.cactus then
+	armor:register_armor("shields:shield_cactus", {
+		description = S("Cactus Shield"),
 		inventory_image = "shields_inv_shield_cactus.png",
 		groups = {armor_shield=5, armor_heal=0, armor_use=1000},
-		wear = 0,
+		armor_groups = {fleshy=5},
+		damage_groups = {cracky=3, snappy=3, choppy=2, crumbly=2, level=1},
+		reciprocate_damage = true,
+		on_damage = function(player, index, stack)
+			play_sound_effect(player, "default_wood_footstep")
+		end,
+		on_destroy = function(player, index, stack)
+			play_sound_effect(player, "default_wood_footstep")
+		end,
 	})
-	minetest.register_tool("shields:shield_enhanced_cactus", {
-		description = "Enhanced Cactus Shield",
+	armor:register_armor("shields:shield_enhanced_cactus", {
+		description = S("Enhanced Cactus Shield"),
 		inventory_image = "shields_inv_shield_enhanced_cactus.png",
-		groups = {armor_shield=6, armor_heal=0, armor_use=500},
-		wear = 0,
+		groups = {armor_shield=1, armor_heal=0, armor_use=1000},
+		armor_groups = {fleshy=8},
```

Comment:
Not the same value for `armor_shield`!

Reply:
Ahh, very sorry for this, didn't realized!